internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -379,6 +379,25 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 
 			"identity": commonschema.SystemOrUserAssignedIdentityOptional(),
 
+			"image_cleaner": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"interval_hours": {
+							Type:     pluginsdk.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
```

Comment:
Depending on the behaviour of the API for `interval_hours` this may need to be two separate top level properties instead of a block. Does it make sense to not provide a value for this property if image cleaner is enabled?

If this does remain a block can we make this consistent with other blocks in this resource? i.e. if the block is present -> the feature is enabled, if it's absent -> feature is disabled.

Reply:
Sure, which one do you prefer, top level or block?

Reply:
It depends on the behaviour of the API, if you enable image cleaner, what default value is set for `interval_hours`, is it a sensible value? And based on the result of that could it make sense to make `interval_hours` required instead? If so then I'd prefer a block, if not then I'd prefer both properties to be flattened to top level.

Reply:
the default value is 48 hours, the value can range between 24 and 2160 hours

Reply:
Thanks for the input, in that case can we flatten these into two top level properties?

Reply:
Yep, will try to make sure it's prepped before next release.